types/index.d.ts

```diff
@@ -0,0 +1,414 @@
+// Type definitions for stampit 4.3
+// Project: https://github.com/stampit-org/stampit, https://stampit.js.org
+// Definitions by: Vasyl Boroviak <https://github.com/koresar>
+//                 Harris Lummis <https://github.com/lummish>
+//                 PopGoesTheWza <https://github.com/popgoesthewza>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="specification.d.ts" />
+
+/**
+ * Return a factory (a.k.a. Stamp) function that will produce new objects using the prototypes that are passed in or composed.
+ * @param options Stampit options object containing methods,
+ * init, props, statics, configurations, and property descriptors.
+ */
+declare function stampit<I = any>(...composables: Array<stampit.StampitComposable<I>>): stampit.Stamp<I>;
+
+declare namespace stampit {
+    // TODO: Add description
+    const version: string;
+    /** The stamp Descriptor */
+    interface Descriptor<I, F extends Specification.Factory<any> = Specification.Factory<I>> extends Specification.Descriptor<I, F> {
+        /** Create a new stamp based on this descriptor */
+        // ? is this function signature trully needed?
+        (...composables: Array<StampitComposable<I>>): Stamp<I>;
+    }
+
+    /** Any composable object (stamp or descriptor) */
+    // ! should have been StampitComposable
+    // type Composable<I> = Stamp<I> | Descriptor<I>;
+
+    // TODO: Add description
+    // ? shouldn't stampit.Option be stampit.(Extended)Descriptor really?
+    interface Options<I, F extends Specification.Factory<any> = Specification.Factory<I>> extends Omit<Specification.Descriptor<I, F>, 'initializers'> {
```

Comment:
Correct. A better name would be `ExtendedDescriptor`

Reply:
Keep in mind, that `ExtendedDescriptor` is only the "input" type. It is always converted to a regular `Descriptor` by stampit.